Scripts/Api.cs

```diff
@@ -123,25 +151,133 @@ public static IEnumerator CreateConversation(string token, Action<string> callba
+
+            yield return request.SendWebRequest();
+
+            if (request.result != UnityWebRequest.Result.Success)
+            {
+                Logger.LogError("Error:" + request.error);
+                yield break;
+            }
+
+            var data = Encoding.UTF8.GetString(request.downloadHandler.data);
+
+            try
+            {
+                var deserialized = JsonConvert.DeserializeObject<GetPlaythroughInfoResponse>(data);
+                callback?.Invoke(deserialized);
+                Logger.Log("Playthrough info request complete");
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Could not generate playthrough information; {e}");
+                throw;
+            }
+
+            // Need to dispose of WebRequest to prevent following error:
+            // "A Native Collection has not been disposed, resulting in a memory leak. Enable Full StackTraces to get more details."
+            request.Dispose();
+        }
+
         /// <summary>
         /// Sets a memory to a specific value in a playthrough.
         /// </summary>
         /// <param name="token">Provide the token of the play-through where the memory should be changed.</param>
-        /// <param name="recallValue">The recall value of the memory.</param>
-        /// <param name="saveValue">The new value of the memory.</param>
-        /// <param name="callback">Called when the mood has successfully been set.</param>
-        public static IEnumerator SetMemory(string token, string recallValue, string saveValue, Action callback = null)
-        {
-            var memoryToSet = new {
-                recallValue = recallValue,
-                saveValue = saveValue,
-            };
-            // This can be multiple memories, but for now this SDK supports setting only one at a time.
-            object[] memoriesToSet = { memoryToSet };
-            object requestParams = new {
+        /// <param name="recallSaveValues">A collection of recall and save values. The key of the Dict acts as the recall value and value acts as the save.</param>
+        /// <param name="callback">Called when the mood has successfully been set.</param>
+        public static IEnumerator SetMemory(string token, Dictionary<string, string> recallSaveValues, Action callback = null)
+        {
+            var count = recallSaveValues.Count;
+
+            Logger.Log($"Setting memories - count: {count}");
+
+            object[] memoriesToSet = new object[count];
+            int i = 0;
+            foreach (var entry in recallSaveValues)
+            {
+                var memoryToSet = new
+                {
+                    recallValue = entry.Key,
+                    saveValue = entry.Value,
+                };
+                Logger.Log($"Setting memory `{memoryToSet.recallValue}` with value `{memoryToSet.saveValue}`...");
+                memoriesToSet[i] = memoryToSet;
+                i++;
+            }

```

Comment:
would this work with a `List` instead of an array? seems more C#-like and simpler

```suggestion
            List<object> memoriesToSet = new List<object>();

            foreach (var entry in recallSaveValues)
            {
                Logger.Log($"Setting memory `{entry.Key}` with value `{entry.Value}`...");
                memoriesToSet.push(new {
                    recallValue = entry.Key,
                    saveValue = entry.Value,
                });
            }

```